lms/djangoapps/courseware/models.py

```diff
@@ -230,3 +230,20 @@ class Meta:
 
     def __unicode__(self):
         return "[OCGLog] %s: %s" % (self.course_id.to_deprecated_string(), self.created)  # pylint: disable=no-member
+
+
+class StudentFieldOverride(models.Model):
+    """
+    Holds the value of a specific field overriden for a student.  This is used
+    by the code in the `courseware.student_field_overrides` module to provide
+    overrides of xblock fields on a per user basis.
+    """
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    location = LocationKeyField(max_length=255, db_index=True)
+    student = models.ForeignKey(User, db_index=True)
+
+    class Meta:   # pylint: disable=missing-docstring
+        unique_together = (('course_id', 'location', 'student'),)
+
+    field = models.CharField(max_length=255)
+    value = models.TextField(default='null')
```

Comment:
I was working on point 3) above, but I ran into an unrelated issue, where south schemamigration doesn't work due to a missing introspection rule for BlockTypeKeyField, which is used in the XModuleStudentPrefsField model. Once I added that, south tells me that the field change to use BlockTypeKeyField is not on a migration yet. I looked at master and it seems to be in the same state. I can fix this, but I should probably do this on a separate edx ticket. What do you think?